src/api-client.ts

```diff
@@ -0,0 +1,114 @@
+import * as httpm from "@actions/http-client";
+
+const userAgent = "GitHub multi-repository variant analysis";
+
+export function getApiClient() {
+  return new httpm.HttpClient(userAgent, [], {
+    allowRetries: true,
+  });
+}
+
+interface InProgressProperties {
+  status: "in_progress";
+}
+
+interface SuccessProperties {
+  status: "succeeded";
+  sourceLocationPrefix: string;
+  resultCount: number;
+  databaseSHA: string;
+}
+
+interface FailureProperties {
+  status: "failed";
+  failureMessage: string;
+}
+
+type updateVariantAnalysisProperties =
+  | InProgressProperties
+  | SuccessProperties
+  | FailureProperties;
+
+export async function setVariantAnalysisRepoInProgress(
+  variantAnalysisId: number,
+  repoId: number
+): Promise<void> {
+  await updateVariantAnalysisStatus(variantAnalysisId, repoId, {
+    status: "in_progress",
+  });
+}
+
+export async function setVariantAnalysisRepoSucceeded(
+  variantAnalysisId: number,
+  repoId: number,
+  sourceLocationPrefix: string,
+  resultCount: number,
+  databaseSHA: string
+): Promise<void> {
+  await updateVariantAnalysisStatus(variantAnalysisId, repoId, {
+    status: "succeeded",
+    sourceLocationPrefix,
+    resultCount,
+    databaseSHA,
+  });
+}
+
+export async function setVariantAnalysisFailed(
+  variantAnalysisId: number,
+  repoId: number,
+  failureMessage: string
+): Promise<void> {
+  await updateVariantAnalysisStatus(variantAnalysisId, repoId, {
+    status: "failed",
+    failureMessage,
+  });
+}
+
+async function updateVariantAnalysisStatus(
+  variantAnalysisId: number,
+  repoId: number,
+  data: updateVariantAnalysisProperties
+): Promise<void> {
+  const http = getApiClient();
+
+  const url = `/codeql/variant-analyses/${variantAnalysisId}/repositories/${repoId}`;
+  const response = await http.patch(url, JSON.stringify(data));
+  if (response.message.statusCode !== 204) {
+    console.log(
+      `Request to ${url} returned status code ${response.message.statusCode}:
+      ${await response.readBody()}`
+    );
+    throw new Error(
+      `Error while setting variant analysis as "${status}". Status code: ${response.message.statusCode}`
```

Comment:
```suggestion
      `Error while setting variant analysis as "${data.status}". Status code: ${response.message.statusCode}`
```

Ooops, missed this one. Turns out we didn't notice and it continued to compile because `status` is a [built-in thing for javascript](https://www.w3schools.com/jsref/prop_win_status.asp), but it's not what we want. If we were running in a browser, we could set some text on the status bar 🤷

Reply:
Woops 🙈 Good spot, thanks!!!